src/AnnotationServiceProvider.php

```diff
@@ -36,14 +36,28 @@ public function boot(Application $app)
         $annotationService = $app['annot'];
 
         // Process annotations for all controllers in given directory
-        if ($app->offsetExists('annot.controllerDir') && strlen($app['annot.controllerDir']) > 0) {
+        if($app->offsetExists('annot.controllerDir') && is_array($app['annot.controllerDir']) && count($app['annot.controllerDir'])) {
+            
+            $controllers = array();
+            foreach($app['annot.controllerDir'] as $controller_dir) {
+                if (!is_dir($controller_dir)) {
+                    throw new RuntimeException("Controller directory: {$controller_dir} does not exist.");
+                }
+                $tmp_controllers = $annotationService->discoverControllers($controller_dir);
+                if(is_array($tmp_controllers) && count($tmp_controllers)) {
+                    $controllers = array_merge($controllers, $tmp_controllers);
+                }                
+            }            
+            $annotationService->registerControllers($controllers);
+            
+        } elseif ($app->offsetExists('annot.controllerDir') && strlen($app['annot.controllerDir']) > 0) {
```

Comment:
There is unnecessary code duplication here.  Instead of having two separate cases here, lets convert the controllerDir value to an array if its a string, then the first case can cover all inputs.